server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/model/vote/IRVBallotInterpretation.java

```diff
@@ -1,25 +1,146 @@
+  @Version
+  private Long version;
 
-package au.org.democracydevelopers.corla.model.vote;
 
-public class IRVBallotInterpretation {
-}
+  /**
+   * The contest to which the interpreted vote belongs.
+   */
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  private Contest contest;
+
+  /**
+   * The record type, either UPLOADED or AUDITOR_ENTERED.
+   */
+  @Column(name = "record_type", nullable = false)
+  @Enumerated(EnumType.STRING)
+  private CastVoteRecord.RecordType recordType;
+
+  /**
+   * ID of the CVR being audited, and whose paper ballot is being interpreted.
+   * There may be multiple records with the same CVR ID, if one ballot contains multiple IRV contests.
+   */
+  @Column(name = "cvr_number", nullable = false)
+  private int cvrNumber;
+
+  /**
+   * The imprinted ID, generally tabulator_id-batch_id-record_id.
+   */
+  @Column(name = "imprinted_id", nullable = false)
+  private String imprintedID;
+
+  /**
+   * List of candidate names with ranks in parentheses representing a raw vote in an IRV contest.
+   * Order is not important.
+   */
+  @Column(name = "raw_choices", columnDefinition = "character varying (1024)", nullable = false)
+  @Convert(converter = StringListConverter.class)
+  private List<String> rawChoices = new ArrayList<>();
+
+  /**
+   * List of candidates names, in order of rank, representing a valid vote in an IRV contest.
+   */
+  @Column(name = "interpretation", columnDefinition = "character varying (1024)", nullable = false)
+  @Convert(converter = StringListConverter.class)
+  private List<String> interpretation = new ArrayList<>();
+
+  /**
+   * Construct an empty IRVBallotInterpretation (for persistence).
+   */
+  public IRVBallotInterpretation() {
+  }
+
+  /**
+   * Create a record of an IRV vote interpretation for a given contest and a given ballot
+   * (identified by the CVR ID).
+   * @param contest        the Contest
+   * @param recordType     the type, expected to be either UPLOADED, AUDITOR_ENTERED, or REAUDITED.
+   * @param cvrNumber      the cvr Number, which appears in the csv file (not to be confused with the cvr_id, which the
+   *                       database makes).
+   * @param imprintedId    the imprinted ID, i.e. tabulator_id-batch_id-record_id.
+   * @param rawChoices     the (invalid) raw IRV choices, e.g. [Bob(1),Alice(3),Chuan(4)].
+   * @param orderedChoices the way colorado-rla interpreted the raw choices, as an order list of names.
+   */
+  public IRVBallotInterpretation(Contest contest, CastVoteRecord.RecordType recordType, int cvrNumber, String imprintedId, List<String> rawChoices, List<String> orderedChoices) {
+    this.contest = contest;
+    this.recordType = recordType;
+    this.cvrNumber = cvrNumber;
+    this.imprintedID = imprintedId;
+    this.rawChoices = rawChoices;
+    this.interpretation = orderedChoices;
+  }
+
+  /**
+   * Output contents as a String appropriate for a log message.
+   * @return the data with headers incorporated.
+   */
```

Comment:
I would probably make the input arguments final. (eg. final cvrNumberHeader etc). I don't know that we have followed this convention consistently to catch bugs where input arguments are accidentally reassigned.

Reply:
Good idea. I've done that for this file and maybe it'd be good to go through the rest too.